napari/resources/_icons.py

```diff
@@ -65,3 +66,103 @@ def get_colorized_svg(
     # use regex to find the svg tag and insert css right after
     # (the '\\1' syntax includes the matched tag in the output)
     return svg_elem.sub(f'\\1{svg_style.format(color, opacity)}', xml)
+
+
+def generate_colorized_svgs(
+    svg_paths: Iterable[Union[str, Path]],
+    colors: Iterable[Union[str, Tuple[str, str]]],
+    opacities: Iterable[float] = (1.0,),
+    theme_override: Optional[Dict[str, str]] = None,
+) -> Iterator[Tuple[str, str]]:
+    """Helper function to generate colorized SVGs.
+
+    This is a generator that yields tuples of ``(alias, icon_xml)`` for every
+    combination (cartesian product) of `svg_path`, `color`, and `opacity`
+    provided. It can be used as input to :func:`_temporary_qrc_file`.
+
+    Parameters
+    ----------
+    svg_paths : Iterable[Union[str, Path]]
+        An iterable of paths to svg files
+    colors : Iterable[Union[str, Tuple[str, str]]]
+        An iterable of colors.  Every icon will be generated in every color. If
+        a `color` item is a string, it should be valid svg color style. Items
+        may also be a 2-tuple of strings, in which case the first item should
+        be an available theme name
+        (:func:`~napari.utils.theme.available_themes`), and the second item
+        should be a key in the theme (:func:`~napari.utils.theme.get_theme`),
+    opacities : Iterable[float], optional
+        An iterable of opacities to generate, by default (1.0,) Opacities less
+        than one can be accessed in qss with the opacity as a percentage
+        suffix, e.g.: ``my_svg_50.svg`` for opacity 0.5.
+    theme_override : Optional[Dict[str, str]], optional
+        When one of the `colors` is a theme ``(name, key)`` tuple,
+        `theme_override` may be used to override the `key` for a specific icon
+        name in `svg_paths`.  For example ``{'exclamation': 'warning'}``, would
+        use the theme "warning" color for any icon named "exclamation.svg" by
+        default `None`
+
+    Yields
+    ------
+    (alias, xml) : Iterator[Tuple[str, str]]
+        `alias` is the name that will used to access the icon in the Qt
+        Resource system (such as QSS), and `xml` is the *raw* colorzied SVG
```

Comment:
```suggestion
        Resource system (such as qss), and `xml` is the *raw* colorzied SVG
```

You seem to mostly be using lowercase for acronyms, so prob best to stick with it.